### PR TITLE
fix memory leak in tfnet

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala [hunk 1]
```diff
@@ -287,18 +295,27 @@ class TFNet(private val graphDef: TFGraphHolder,
       emptyTFTensorArray(outputs.asScala.slice(0, outputNames.length))
       // tempTensors will be cleaned up after backward
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: if (!this.isTraining()){
    tensorManager.destructTFTensors()
}
else {
  emptyTFTensorArray(tempTFTensors)
  emptyTFTensorArray(weightTFTensors)
}

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala [hunk 2]
```diff
@@ -402,6 +419,12 @@ class TFNet(private val graphDef: TFGraphHolder,
         tensorManager.destructTFTensors()
         throw ex
     }
+
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: just call `tensorManager.destructTFTensors()` in the try block?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala (hunk 1 above)
Comment (yangw1234) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: fixed

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala (hunk 2 above)
Comment (yangw1234) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: changed to calling `tensorManager.destructTFTensors()` in finally

**Comment by yangw1234** [comment]
jenkins pass http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2287/

**Pull request merged**